ospfd/ospf_vty.c

```diff
@@ -8165,234 +8716,240 @@ static int config_write_interface(struct vty *vty)
 	int write = 0;
 	struct route_node *rn = NULL;
 	struct ospf_if_params *params;
-	struct ospf *ospf = ospf_lookup();
+	struct ospf *ospf = NULL;
+	struct listnode *node = NULL;
 
-	for (ALL_LIST_ELEMENTS_RO(vrf_iflist(VRF_DEFAULT), n1, ifp)) {
-		if (memcmp(ifp->name, "VLINK", 5) == 0)
-			continue;
+	/* Traverse all ospf [vrf] instances */
+	for (ALL_LIST_ELEMENTS_RO(om->ospf, node, ospf)) {
 
-		if (ifp->ifindex == IFINDEX_DELETED)
-			continue;
+		for (ALL_LIST_ELEMENTS_RO(vrf_iflist(ospf->vrf_id), n1, ifp)) {
+			if (memcmp(ifp->name, "VLINK", 5) == 0)
+				continue;
 
```

Comment:
please create a sub-function here to print the config, i.e. `config_write_interface_one(struct vty *vty, struct ospf *ospf, struct ospf_if_params *params)`

(to reduce the nesting depth for a long function)

Reply:
created sub-function.